File: src/2154_KeepMultiplyingFoundValuesByTwo/2154.scala
object Solution {
    def findFinalValue(nums: Array[Int], original: Int): Int = {
       val tally = nums.toSet
       def loop(original: Int): Int = {
            tally.contains(original) match {
                case true  => loop(original*2)
                case false => original
            } 
       }

       loop(original)
    }
}

File: src/2154_KeepMultiplyingFoundValuesByTwo/2154.go
func findFinalValue(nums []int, original int) int {
    lookup := make(map[int]bool)
    for _, n := range nums{
        lookup[n] = true 
    }
    
    for _, ok := lookup[original] ; ok ; {
        original *= 2
        _, ok = lookup[original]
    }

    return original 
}